test/podFailover/controller/main.go

```diff
@@ -199,6 +204,24 @@ func deleteStorageClass(ctx context.Context, clientset *kubernetes.Clientset, na
 	}
 }
 
+func deleteDeployments(ctx context.Context, clientset *kubernetes.Clientset, deployments ...*apps.Deployment) {
+	for _, deployment := range deployments {
+		err := clientset.AppsV1().Deployments(podFailoverNamespace).Delete(ctx, deployment.Name, metav1.DeleteOptions{})
```

Comment:
This marks deployments for deleting but doesn't actually wait for the pods to be deleted. In practice, the `Deployment` object is deleted almost immediately, but it takes a non-trivial amount of time for the deployment's `Pod` objects to be deleted. Should we wait for that clean-up to complete?

Reply:
Yeah, it is cleaner to wait for the pods  to be deleted too. Made changes to handle that.